src/main/java/org/fcrepo/client/FcrepoResponse.java

```diff
@@ -55,6 +72,36 @@ public FcrepoResponse(final URI url, final int statusCode,
     }
 
     /**
+     * {@inheritDoc}
+     * <p>
+     * Implementation note: Invoking this method will close the underlying {@code InputStream} containing the entity
+     * body of the HTTP response.
+     * </p>
+     *
+     * @throws IOException if there is an error closing the underlying HTTP response stream.
+     */
+    @Override
+    public void close() throws IOException {
+        if (!this.closed && this.body != null) {
+            try {
+                this.body.close();
+            } finally {
+                this.closed = true;
+            }
+        }
+    }
+
+    /**
+     * Whether or not the resources have been freed from this response.  There should be no expectation that a closed
+     * response provides access to the {@link #getBody() entity body}.
+     *
+     * @return {@code true} if resources have been freed, otherwise {@code false}
+     */
+    public boolean isClosed() {
```

Comment:
My thought was allowing a client to avoid accessing the underlying closed response (or exhausted `InputStream`), which may result in an exception.  Not wedded to it, though, and whether or not this needs to be exposed may be influenced by factoring up the `CloseableHttpResponse`.